tendrl/node_agent/flows/import_cluster/__init__.py

```diff
@@ -42,3 +44,16 @@ def run(self):
             import_ceph(tendrl_ns.tendrl_context.integration_id)
         else:
             import_gluster(tendrl_ns.tendrl_context.integration_id)
+
+        # import cluster's run() should not return unless the new cluster entry
+        # is updated in etcd, as the job is marked as finished if this
+        # function is returned. This might lead to inconsistancy in the API
+        # functionality. The below loop waits for the cluster details
+        # to be updated in etcd.
+        while True:
+            gevent.sleep(2)
+            try:
+                tendrl_ns.etcd_orm.client.read("/clusters/%s" % integration_id)
+                break
+            except etcd.EtcdKeyNotFound:
+                continue
```

Comment:
The common job handler expects a flow_execution_failed exception and marks job as failed if this exception is raised in the flow. In this case, since we have stop gap gluster and ceph import flows they are not raised. When they are implemented properly they will be raising the exception when some issue occures